src/health_misinfo_shared/youtube_api.py

```diff
@@ -104,18 +104,32 @@ def load_texts(folder) -> list[dict]:
     return flat_list
 
 
-def form_chunks(transcript: list[dict]) -> Iterator[str]:
-    """Split/merged a list of sentences into series of overlapping text chunks."""
+def form_chunks(transcript_obj: dict) -> Iterator[dict]:
+    """Split/merged a list of sentences into series of overlapping text chunks.
+    Each chunk is a dict containing the text and the start/end timestamps"""
     current_chunk_text = ""
-    for sentence in transcript:
-        current_chunk_text += sentence["sentence_text"] + " "
-        if len(current_chunk_text) > 5000:
-            yield current_chunk_text
+    current_chunk_start_offset = 0.0
+    current_chunk_end_offset = 0.0
+    for s in transcript_obj:
```

Comment:
This is minor, but I think `sentence` is a clearer variable name than `s`.

Reply:
Agreed. I'll fix.